apps/server/src/modules/learnroom/uc/course-info.uc.ts

```diff
@@ -0,0 +1,118 @@
+import { AuthorizationContextBuilder, AuthorizationService } from '@modules/authorization';
+import { ClassService } from '@modules/class';
+import { Group, GroupService } from '@modules/group';
+import { School, SchoolService } from '@modules/school';
+import { UserService } from '@modules/user';
+import { Injectable } from '@nestjs/common';
+import { Page, UserDO } from '@shared/domain/domainobject';
+import { User } from '@shared/domain/entity';
+import { IFindOptions, Pagination, Permission, SortOrder, SortOrderMap } from '@shared/domain/interface';
+import { EntityId } from '@shared/domain/types';
+import { Course as CourseDO } from '../domain';
+import { CourseFilter, CourseSortQueryType, CourseStatusQueryType } from '../domain/interface';
+import { CourseDoService } from '../service';
+import { CourseInfoDto } from './dto';
+
+@Injectable()
+export class CourseInfoUc {
+	public constructor(
+		private readonly authService: AuthorizationService,
+		private readonly schoolService: SchoolService,
+		private readonly courseDoService: CourseDoService,
+		private readonly groupService: GroupService,
+		private readonly userService: UserService,
+		private readonly classService: ClassService
+	) {}
+
+	public async getCourseInfo(
+		userId: EntityId,
+		schoolId: EntityId,
+		sortByField: CourseSortQueryType = CourseSortQueryType.NAME,
+		courseStatusQueryType?: CourseStatusQueryType,
+		pagination?: Pagination,
+		sortOrder: SortOrder = SortOrder.asc
+	): Promise<Page<CourseInfoDto>> {
+		const school: School = await this.schoolService.getSchoolById(schoolId);
+
+		const user: User = await this.authService.getUserWithPermissions(userId);
+		this.authService.checkPermission(user, school, AuthorizationContextBuilder.read([Permission.ADMIN_VIEW]));
+
+		const order: SortOrderMap<CourseDO> = { [sortByField]: sortOrder };
+		const filter: CourseFilter = { schoolId, courseStatusQueryType };
+		const options: IFindOptions<CourseDO> = { pagination, order };
+		const courses: Page<CourseDO> = await this.courseDoService.getCourseInfo(filter, options);
+		let page: Page<CourseInfoDto> = new Page<CourseInfoDto>([], courses.total);
+
+		if (courses.total === 0) {
+			return page;
+		}
```

Comment:
you don't need this part. in getCourseData you use the .map function which returns an empty array, when it operates on an empty array